### PR TITLE
Introduce mdbook

### LINKED ISSUE #3: Use mdbook?
Seems https://github.com/rust-lang-nursery/mdBook is a nice alternative to Gitbook in Rust.

### PR DESCRIPTION
**Pull request by Hoverbear**
Based off #2 .
Fixes #3.

Introduces `mdbook`, which is used by the Rust project for their books.

We can have this build and then be hosted by Github Pages very easily.

It also introduces a CI script that ~should~ build the book and make sure the committer has committed the newest build.

After merging please:

* Enable Github pages on this repo for the master branch.

### CODE REVIEW

**Comment by QueenyJin** [comment]
@CaitinChen PTAL.

File: README.md [hunk 1]
```diff
@@ -6,13 +6,18 @@
 - The detailed designs for different hierarchy layers of TiKV
 - The techniques we choose to build TiKV
 - Any problem we meet when we build TiKV and how we solve them
-- How you can build your own services based on TiKV
+- How can you build your own services based on TiKV
```
Comment (CaitinChen) [inline README.md]: ->**How to** build your own services based on TiKV

File: src/overview/introduction.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+TiKV is fully [ACID](https://en.wikipedia.org/wiki/ACID_(computer_science)) compliant and features automatic horizontal scalability, global data consistency, geo-replication, and many other features. It can be used as a building block for other high-level services. For example, we have already used TiKV to support [TiDB](https://github.com/pingcap/tidb) - a next-generation [HTAP](https://en.wikipedia.org/wiki/Hybrid_transactional/analytical_processing_(HTAP)) database.
+
+ In this book, we will introduce everything about TiKV, including why we built it and how we continue to improve it, what problems we have met, what the core technologies are and why, etc. We hope that through this book, you can develop a deep understanding of TiKV, build your knowledge of distributed programming, or even get inspired to build your own distributed system. :-)
```
Comment (CaitinChen) [inline src/overview/introduction.md]: Please delete the space before this paragraph.

File: src/overview/introduction.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+## History 
+
+In the middle of 2015, we decided to build a database which solved the scaling problem of MySQL. At that time, the most used way was to build a proxy on top of the MySQL servers, we didn't (and still don't!) think a proxy was an elegant way.
```
Comment (CaitinChen) [inline src/overview/introduction.md]: ... the most used way was to build a proxy on top of the MySQL servers, **but** we didn't (and still don't!) think a proxy was an elegant way.

**Comment by siddontang** [comment]
@Hoverbear how to visist this mdbook? through head-first-tikv.github.com?

**Comment by Hoverbear** [comment]
@siddontang You will need to look locally. Since this is not merged, and github pages is not activated, Github is not hosting it.

You can try on the branch with `cargo install mdbook && mdbook serve` then visiting `localhost:3000`.

**Comment by siddontang** [comment]
@Hoverbear 

Please address the comments.

Rest LGTM

**Comment by Hoverbear** [comment]
:rofl: It seems the content reviewed in #2 needed more review. :)

**Comment by siddontang** [comment]
PTAL @tikv/pingcap-i18n

File: src/overview/introduction.md [hunk 4]
```diff
@@ -6,32 +6,32 @@ TiKV is fully [ACID](https://en.wikipedia.org/wiki/ACID_(computer_science)) comp
...
 
 ## Architecture
 
-Following is the architecture of TiKV: 
+Following is the architecture of TiKV:
 
 ![Architecture](./architecture.png)
 
-According to the picture, there are three TiKV instances in the cluster and each instance uses one [RocksDB](https://github.com/facebook/rocksdb) to save data. On top of RocksDB, we use [Raft](https://raft.github.io/) consensus algorithm to replicate the data. Mostly, we should use at least three replicas to keep data safe and consistent, and these replicas form a Raft group. 
+According to the picture, there are three TiKV instances in the cluster and each instance uses one [RocksDB](https://github.com/facebook/rocksdb) to save data. On top of RocksDB, we use [Raft](https://raft.github.io/) consensus algorithm to replicate the data. Mostly, we should use at lease three replicas to keep data safety and consistence, and these replicas form a Raft group.
```
Comment (siddontang) [inline src/overview/introduction.md]: at least

Comment (Hoverbear) [reply src/overview/introduction.md]: I'm very confused. I do not recall editing this file when I made the PR... Thank for you finding this.

**Pull request merged**